polyfill/intersection-observer-test.js

```diff
@@ -941,6 +941,652 @@ describe('IntersectionObserver', function() {
+          return target;
+        }
+        iframeTargetEl1 = createTarget('target1', 'blue');
+        iframeTargetEl2 = createTarget('target2', 'green');
+      });
+    });
+
+    afterEach(function() {
+      rootEl.removeChild(iframe);
+    });
+
+    function rect(r) {
+      return {
+        top: r.top,
+        left: r.left,
+        width: r.width != null ? r.width : r.right - r.left,
+        height: r.height != null ? r.height : r.bottom - r.top,
+        right: r.right != null ? r.right : r.left + r.width,
+        bottom: r.bottom != null ? r.bottom : r.top + r.height
+      };
+    }
+
+    function getRootRect(doc) {
+      var html = doc.documentElement;
+      var body = doc.body;
+      return rect({
+        top: 0,
+        left: 0,
+        right: html.clientWidth || body.clientWidth,
+        width: html.clientWidth || body.clientWidth,
+        bottom: html.clientHeight || body.clientHeight,
+        height: html.clientHeight || body.clientHeight
+      });
+    }
+
+    it('iframe targets do not intersect with a top root element', function(done) {
+      var io = new IntersectionObserver(function(unsortedRecords) {
+        var records = sortRecords(unsortedRecords);
+        expect(records.length).to.be(2);
+        expect(records[0].isIntersecting).to.be(false);
+        expect(records[1].isIntersecting).to.be(false);
+        done();
+        io.disconnect();
+      }, {root: rootEl});
+      io.observe(iframeTargetEl1);
+      io.observe(iframeTargetEl2);
+    });
+
+    it('triggers for all targets in top-level root', function(done) {
+      var io = new IntersectionObserver(function(unsortedRecords) {
+        var records = sortRecords(unsortedRecords);
+        expect(records.length).to.be(2);
+        expect(records[0].isIntersecting).to.be(true);
+        expect(records[0].intersectionRatio).to.be(1);
+        expect(records[1].isIntersecting).to.be(false);
+        expect(records[1].intersectionRatio).to.be(0);
+
+        // The rootBounds is for the document's root.
+        expect(records[0].rootBounds.height).to.be(innerHeight);
+
+        done();
+        io.disconnect();
+      });
+      io.observe(iframeTargetEl1);
+      io.observe(iframeTargetEl2);
+    });
+
+    it('triggers for all targets in iframe-level root', function(done) {
+      var io = new IntersectionObserver(function(unsortedRecords) {
+        var records = sortRecords(unsortedRecords);
+        expect(records.length).to.be(2);
+        expect(records[0].intersectionRatio).to.be(1);
+        expect(records[1].intersectionRatio).to.be(1);
+
+        // The rootBounds is for the document's root.
+        expect(rect(records[0].rootBounds)).
+          to.eql(rect(iframeDoc.body.getBoundingClientRect()));
+
+        done();
+        io.disconnect();
+      }, {root: iframeDoc.body});
+      io.observe(iframeTargetEl1);
+      io.observe(iframeTargetEl2);
+    });
+
+    it('calculates rects for a fully visible frame', function(done) {
```

Comment:
Can you confirm that this test is passing for you? I see failures when running it both with and without the polyfill. (Note: I see the same for the tests whose description starts with "calculates..."):

<img width="951" alt="Screen Shot 2020-04-08 at 5 44 12 PM" src="https://user-images.githubusercontent.com/326742/78846453-aeb4fd00-79c0-11ea-97ba-800e6c18de2c.png">

Reply:
This is very odd. And especially w.r.t. to `right: 85`. But I can confirm that all tests pass for me for both native and polyfill and on all the browsers that I've tried. And I've tested on Chrome/Firefox/Safari. Having difficulties testing on IE at the moment, but will certainly find a way to test IE before merging (and any advise would be appreciated).

Reply:
Turns out, this is not odd at all. Just the test wasn't accounting for the scrollbar. I fixed it. And tested on IE11 and IE10.